From 1f35681ffdfd70292281efc8c3fbe31687decd1b Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:24:31 -0400
Subject: [pci] portdrv: Cleanup error paths

Message-id: <20140719042430.19438.2887.stgit@gir.stowe>
Patchwork-id: 85979
O-Subject: [RHEL7.1 PATCH 17/81] PCI/portdrv: Cleanup error paths
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 8f3acca9acec1503f6b374faef2d1013cbf502af

commit 8f3acca9acec1503f6b374faef2d1013cbf502af
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Thu Dec 19 14:20:09 2013 -0700

    PCI/portdrv: Cleanup error paths

    Make the straightline path the normal no-error path.  Check for errors and
    return them directly, instead of checking for success and putting the
    normal path in an "if" body.

    No functional change.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/pcie/portdrv_core.c b/drivers/pci/pcie/portdrv_core.c
index ce9d9ae..806aa91 100644
--- a/drivers/pci/pcie/portdrv_core.c
+++ b/drivers/pci/pcie/portdrv_core.c
@@ -344,11 +344,13 @@ static int pcie_device_init(struct pci_dev *pdev, int service, int irq)
  device_enable_async_suspend(device);
 
  retval = device_register(device);
- if (retval)
+ if (retval) {
   kfree(pcie);
- else
-  get_device(device);
- return retval;
+  return retval;
+ }
+
+ get_device(device);
+ return 0;
 }
 
 /**
@@ -498,12 +500,12 @@ static int pcie_port_probe_service(struct device *dev)
 
  pciedev = to_pcie_device(dev);
  status = driver->probe(pciedev);
- if (!status) {
-  dev_printk(KERN_DEBUG, dev, "service driver %s loaded\n",
-   driver->name);
-  get_device(dev);
- }
- return status;
+ if (status)
+  return status;
+
+ dev_printk(KERN_DEBUG, dev, "service driver %s loaded\n", driver->name);
+ get_device(dev);
+ return 0;
 }
 
 /**
-- 
1.7.1